src/ristretto/constants.rs

```diff
@@ -55,24 +57,27 @@ pub const RISTRETTO_NUMS_POINTS_COMPRESSED: [CompressedRistretto; 10] = [
 
 lazy_static! {
     /// A static array of pre-generated NUMS points
-    pub static ref RISTRETTO_NUMS_POINTS: [RistrettoPoint; 10] = {
-        let mut arr = [RistrettoPoint::default(); 10];
-        for i in 0..10 {
+    pub static ref RISTRETTO_NUMS_POINTS: [RistrettoPoint; NUMBER_NUMS_POINTS] = {
+        let mut arr = [RistrettoPoint::default(); NUMBER_NUMS_POINTS];
+        for i in 0..NUMBER_NUMS_POINTS {
             arr[i] = RISTRETTO_NUMS_POINTS_COMPRESSED[i].decompress().unwrap();
         }
         arr
     };
+
+    /// Precomputation table for the first point, which is used as the default commitment generator
+    pub static ref RISTRETTO_NUMS_TABLE_0: RistrettoBasepointTable = RistrettoBasepointTable::create(&RISTRETTO_NUMS_POINTS[0]);
```

Comment:
If I understand correctly, we only need the first entry of `RISTRETTO_NUMS_POINTS` to defined `RISTRETTO_NUM_TABLE_0`, why do we need then an array `RISTRETTO_NUMS_POINTS_COMPRESSED` of len `10` ?

Reply:
The library already contained this set of points for implementers to use. The default Pedersen factory uses the standard base point `G` (from the upstream curve library) and the first point in the set as its generators. The [extended Pedersen factories](https://github.com/tari-project/tari-crypto/blob/main/src/ristretto/pedersen/extended_commitment_factory.rs) use additional points from the set.

An earlier version of this PR generated precomputation tables for all the points in the set, but it was decided this was wasted computation at this point, since only the default Pedersen factory is used in the codebase (and the extended Pedersen factories use multiscalar multiplication for commitment evaluation, which would require different precomputation table structures and architecture).

Reply:
@CjS77 had mentioned elsewhere that a future strategy could be to keep some expensive curve point generation operations (like hash-based generation and precomputation) within the commitment factories, and then ensure that these factories are generated only once within the implementing codebase. This is already how the much larger set of range proof generators is handled. However, default Pedersen factory copies are currently generated on the fly more often, presumably because it was cheap to do so.